rmw_connextdds_common/src/common/rmw_config.cpp

```diff
@@ -219,3 +219,23 @@ RMW_Connext_Config::find_type_transformation(const std::string & type_transform_
     return nullptr;
   }
 }
+
+rmw_ret_t
+RMW_Connext_Config::notify_type_use(DDS_TypeCode * type_pointer)
+{
+  if (type_counter_.find(type_pointer) == type_counter_.end())
+    type_counter_[type_pointer] = 1;
+  else
+    type_counter_[type_pointer] += 1;
+
+  return RMW_RET_OK;
+}
```

Comment:
You should always use braces (`{ }`) even with one line blocks. Also, you should cache the result of searching for the key, so you don't have to search for it again in order to update it. Since the function is called within a try/catch block, we can make it return `void` and let it propagate any exception to the caller, which will "turn them into errors".

```suggestion
void
RMW_Connext_Config::on_network_type_registered(const std::string & type_name)
{
  auto existing_type = network_types_.find(type_name);
  if (existing_type == network_types_.end()) {
    network_types_[type_name] = 1;
  } else {
    *existing_type += 1;
  }
}
```